Plugins/GeoModel/src/GeoModelDetectorObjectFactory.cpp

```diff
@@ -0,0 +1,219 @@
+  std::vector<GeoChildNodeWithTrf> sensitives;
+  for (auto subvolume : subvolumes) {
+    if (matches(subvolume.nodeName, subvolume.volume)) {
+      sensitives.push_back(subvolume);
+    }
+    std::vector<GeoChildNodeWithTrf> senssubsubvolumes =
+        findAllSubVolumes(subvolume.volume);
+    std::transform(std::make_move_iterator(senssubsubvolumes.begin()),
+                   std::make_move_iterator(senssubsubvolumes.end()),
+                   std::back_inserter(sensitives),
+                   [&subvolume](GeoChildNodeWithTrf &&volume) {
+                     volume.transform = subvolume.transform * volume.transform;
+                     return volume;
+                   });
+  }
+  return sensitives;
+}
+bool Acts::GeoModelDetectorObjectFactory::convertBox(std::string name) {
+  auto convB = std::any_of(
+      m_cfg.convertBox.begin(), m_cfg.convertBox.end(),
+      [&](const auto &n) { return name.find(n) != std::string::npos; });
+  // return (name.find(m_cfg.convertBox) != std::string::npos);
+  return convB;
+}
+void Acts::GeoModelDetectorObjectFactory::convertFpv(
+    std::string name, GeoFullPhysVol *fpv, Cache &cache,
+    const GeometryContext &gctx) {
+  PVConstLink physVol{fpv};
+
+  // get children
+  std::vector<GeoChildNodeWithTrf> subvolumes =
+      getChildrenWithRef(physVol, false);
+  if (subvolumes.size() > 0) {
+    // vector containing all subvolumes to be converted to surfaces
+    std::vector<GeoChildNodeWithTrf> surfaces = findAllSubVolumes(physVol);
+    std::vector<GeoModelSensitiveSurface> sensitives;
+
+    for (auto surface : surfaces) {
+      const Transform3 &transform =
+          fpv->getAbsoluteTransform() * surface.transform;
+      convertSensitive(surface.volume, transform, sensitives);
+    }
+    cache.sensitiveSurfaces.insert(cache.sensitiveSurfaces.end(),
+                                   sensitives.begin(), sensitives.end());
+    if (convertBox(name)) {
+      const GeoLogVol *logVol =
+          physVol->getLogVol();  // get logVol for the shape of the volume
+      const GeoShape *shape = logVol->getShape();  // get shape
+      const Acts::Transform3 &fpvtransform = fpv->getAbsoluteTransform(nullptr);
+
+      // convert bounding boxes with surfaces inside
+      std::shared_ptr<Experimental::DetectorVolume> box =
+          Acts::GeoModel::convertVolume(gctx, shape, name, fpvtransform,
+                                        sensitives);
+      cache.boundingBoxes.push_back(box);
+    }
+  } else {
+    // convert fpvs to surfaces
+    const Transform3 &transform = fpv->getAbsoluteTransform();
+    convertSensitive(fpv, transform, cache.sensitiveSurfaces);
+  }
+}
+// lambda to determine if object fits query
+bool Acts::GeoModelDetectorObjectFactory::matches(const std::string &name,
+                                                  PVConstLink physvol) {
+  if (m_cfg.nameList.empty() && m_cfg.materialList.empty()) {
+    return true;
```

Comment:
Shouldn't this be `false`? If there's no matching, we take everything?

Reply:
The idea is that if nothing is passed we assume that everything should be converted so I think this is correct.